Attribute name with colon can't be transform in XML file type

When do transform with applicationsettings with the attribute key that contain colon, it will error. 

Version: 1.2.0

Sample config 
```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <appSettings>
    <add key="blog:theme" value="green" />
  </appSettings>
</configuration>
```

Sample transform config 

```
{
  "configuration/appSettings/add[@key='blog:theme']/@value": "red",
}

```
Error :
```
2017-11-14T11:04:04.9849036Z ##[error]System.Management.Automation.MethodInvocationException: Exception calling "Transform" with "4" argument(s): "The ''' character, hexadecimal value 0x27, cannot be included in a name." ---> System.Xml.XmlException: The ''' character, hexadecimal value 0x27, cannot be included in a name.
2017-11-14T11:04:04.9849036Z    at System.Xml.XmlConvert.VerifyNCName(String name, ExceptionType exceptionType)
2017-11-14T11:04:04.9849036Z    at System.Xml.Linq.XName..ctor(XNamespace ns, String localName)
2017-11-14T11:04:04.9849036Z    at System.Xml.Linq.XNamespace.GetName(String localName, Int32 index, Int32 count)
2017-11-14T11:04:04.9982685Z    at MagicChunks.Helpers.XmlExtensions.GetNameWithNamespace(String name, XElement element, String defaultNamespace)
2017-11-14T11:04:04.9982685Z    at MagicChunks.Helpers.XmlExtensions.<>c__DisplayClass1_0.<GetChildElementByName>b__0(XElement e)
2017-11-14T11:04:04.9982685Z    at System.Linq.Enumerable.FirstOrDefault[TSource](IEnumerable`1 source, Func`2 predicate)
2017-11-14T11:04:04.9982685Z    at MagicChunks.Documents.XmlDocument.FindPath(IEnumerable`1 path, XElement current, String documentNamespace)
2017-11-14T11:04:04.9982685Z    at MagicChunks.Documents.XmlDocument.ReplaceKey(String[] path, String value)
2017-11-14T11:04:04.9982685Z    at MagicChunks.Core.Transformer.Transform(IDocument source, TransformationCollection transformations)
2017-11-14T11:04:04.9982685Z    at MagicChunks.TransformTask.Transform(String type, String sourcePath, String targetPath, TransformationCollection transformation)
2017-11-14T11:04:04.9982685Z    at CallSite.Target(Closure , CallSite , Type , Object , Object , Object , Object )
2017-11-14T11:04:04.9982685Z    --- End of inner exception stack trace ---
2017-11-14T11:04:04.9982685Z    at System.Management.Automation.ExceptionHandlingOps.CheckActionPreference(FunctionContext funcContext, Exception exception)
2017-11-14T11:04:04.9982685Z    at System.Management.Automation.Interpreter.ActionCallInstruction`2.Run(InterpretedFrame frame)
2017-11-14T11:04:04.9982685Z    at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
2017-11-14T11:04:04.9982685Z    at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
```

--- a/src/MagicChunks.Tests/Documents/XmlDocumentTests.cs
+++ b/src/MagicChunks.Tests/Documents/XmlDocumentTests.cs
@@ -648,5 +648,37 @@ namespace MagicChunks.Tests.Documents
   </connectionStrings>
 </configuration>", result, ignoreCase: true, ignoreLineEndingDifferences: true, ignoreWhiteSpaceDifferences: true);
         }
+
+        [Fact]
+        public void TransformWithColonInAttributeName()
+        {
+            // Arrange
+            var document = new XmlDocument(@"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+  <appSettings>
+    <add key=""Auth:Environment"" value=""Testing"" />
+    <add key=""Auth:Credential"" value=""abcefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"" />
+    <add key=""Auth:Type"" value=""Basic"" />
+  </appSettings>
+</configuration>");
+            
+            document.ReplaceKey(new[] { "configuration", "appSettings", "add[@key = 'Auth:Environment']", "@value" }, "Production");
+            document.ReplaceKey(new[] { "configuration", "appSettings", "add[@key=\"Auth:Credential\"]", "@value" }, "XqoL6MXqOPeGTxXrJsQA7gK5cXLvnlSYJUtAwFWbRo7GDkbsSu");
+            document.ReplaceKey(new[] { "configuration", "appSettings", "add[@key=\"Auth:Type\"]", "@value" }, "Bearer");
+
+            var result = document.ToString();
+
+
+            // Assert
+            Assert.Equal(@"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+  <appSettings>
+    <add key=""Auth:Environment"" value=""Production"" />
+    <add key=""Auth:Credential"" value=""XqoL6MXqOPeGTxXrJsQA7gK5cXLvnlSYJUtAwFWbRo7GDkbsSu"" />
+    <add key=""Auth:Type"" value=""Bearer"" />
+  </appSettings>
+</configuration>", result, ignoreCase: true, ignoreLineEndingDifferences: true, ignoreWhiteSpaceDifferences: true);
+        }
+
     }
 }

--- a/src/MagicChunks.Tests/Documents/XmlDocumentTests.cs
+++ b/src/MagicChunks.Tests/Documents/XmlDocumentTests.cs
@@ -659,12 +659,14 @@ namespace MagicChunks.Tests.Documents
     <add key=""Auth:Environment"" value=""Testing"" />
     <add key=""Auth:Credential"" value=""abcefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"" />
     <add key=""Auth:Type"" value=""Basic"" />
+    <add key=""Auth:Provider:Name"" value=""Name"" />
   </appSettings>
 </configuration>");
             
             document.ReplaceKey(new[] { "configuration", "appSettings", "add[@key = 'Auth:Environment']", "@value" }, "Production");
             document.ReplaceKey(new[] { "configuration", "appSettings", "add[@key=\"Auth:Credential\"]", "@value" }, "XqoL6MXqOPeGTxXrJsQA7gK5cXLvnlSYJUtAwFWbRo7GDkbsSu");
             document.ReplaceKey(new[] { "configuration", "appSettings", "add[@key=\"Auth:Type\"]", "@value" }, "Bearer");
+            document.ReplaceKey(new[] { "configuration", "appSettings", "add[@key=\"Auth:Provider:Name\"]", "@value" }, "ABC");
 
             var result = document.ToString();
 
@@ -676,6 +678,7 @@ namespace MagicChunks.Tests.Documents
     <add key=""Auth:Environment"" value=""Production"" />
     <add key=""Auth:Credential"" value=""XqoL6MXqOPeGTxXrJsQA7gK5cXLvnlSYJUtAwFWbRo7GDkbsSu"" />
     <add key=""Auth:Type"" value=""Bearer"" />
+    <add key=""Auth:Provider:Name"" value=""ABC"" />
   </appSettings>
 </configuration>", result, ignoreCase: true, ignoreLineEndingDifferences: true, ignoreWhiteSpaceDifferences: true);
         }

--- a/src/MagicChunks/Helpers/XmlExtensions.cs
+++ b/src/MagicChunks/Helpers/XmlExtensions.cs
@@ -43,7 +43,7 @@ namespace MagicChunks.Helpers
                 result = XName.Get(name, defaultNamespace);
             else
             {
-                var attributeNameParts = name.Split(':');
+                var attributeNameParts = name.Split(':');                
                 var attributeNamespace = element.GetNamespaceOfPrefix(attributeNameParts[0]);
                 if (attributeNamespace != null)
                     result = XName.Get(attributeNameParts[1], attributeNamespace.NamespaceName);
@@ -58,8 +58,10 @@ namespace MagicChunks.Helpers
         {
             if (!NodeIndexEndingRegex.IsMatch(name))
             {
+                bool isElementNameWithNamespace = name.IndexOf(':') > 0 && !(name.Split(':')[0].Contains("'") || (name.Split(':')[0].Contains(@"""")));
+
                 return source?.Elements()
-                    .FirstOrDefault(e => name.IndexOf(':') == -1 ?
+                    .FirstOrDefault(e => !isElementNameWithNamespace ?
                                             String.Compare(e.Name.LocalName, name, StringComparison.OrdinalIgnoreCase) == 0 :
                                             e.Name == name.GetNameWithNamespace(source, String.Empty));
             }

